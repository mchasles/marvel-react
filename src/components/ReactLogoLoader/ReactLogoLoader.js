import React from 'react';
import './ReactLogoLoader.scss';
import logo from './logo.svg';

export const ReactLogoLoader = () => {
  return (<img className="react-logo-loader" src={logo} alt="React logo spinning" />);
}