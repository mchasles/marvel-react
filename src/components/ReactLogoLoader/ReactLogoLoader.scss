.react-logo-loader {
  position: absolute;
  top: 50%;
  left: 50%;
  animation: react-logo-loader-spin infinite 1s ease-in-out;
  height: 180px;
  width: 180px;
}

@keyframes react-logo-loader-spin {
  from { transform: translate(-50%, -50%) rotate(0deg) ; }
  to { transform: translate(-50%, -50%) rotate(360deg); }
}