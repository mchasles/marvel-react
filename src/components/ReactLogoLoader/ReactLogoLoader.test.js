import React from 'react';
import ReactDOM from 'react-dom';
import {ReactLogoLoader} from './ReactLogoLoader';

import renderer from 'react-test-renderer';

it('renders ReactLogoLoader without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ReactLogoLoader />, div);
});

it('renders ReactLogoLoader correctly when there is no character prop set', () => {
  const list = renderer.create(<ReactLogoLoader />).toJSON();
  expect(list).toMatchSnapshot();
});
