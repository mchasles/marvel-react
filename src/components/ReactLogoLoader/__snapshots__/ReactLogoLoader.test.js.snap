exports[`test renders ReactLogoLoader correctly when there is no character prop set 1`] = `
<img
  alt="React logo spinning"
  className="react-logo-loader"
  src="logo.svg" />
`;
